extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -1139,21 +1140,27 @@ private PushdownResult pushdownDataIndex(
             final PushdownResult result) {
         final RowSetBuilderRandom matchingBuilder = RowSetFactory.builderRandom();
         try (final SafeCloseable ignored = LivenessScopeStack.open()) {
-            final WhereFilter copiedFilter = filter.copy();
-            copiedFilter.init(dataIndex.table().getDefinition());
-
-            // TODO: When https://deephaven.atlassian.net/browse/DH-19443 is implemented, we should be able
-            // to use the filter directly on the index table without renaming.
-            final Collection<Pair> renamePairs = renameMap.entrySet().stream()
-                    .map(entry -> Pair.of(ColumnName.of(entry.getValue()),
-                            ColumnName.of(entry.getKey())))
-                    .collect(Collectors.toList());
-            final Table renamedIndexTable = dataIndex.table().renameColumns(renamePairs);
-
+            final long threshold = (long) (dataIndex.table().size() / QueryTable.DATA_INDEX_FOR_WHERE_THRESHOLD);
+            if (result.maybeMatch().size() <= threshold) {
+                return result.copy();
+            }
+            // Extract the fundamental filter, ignoring barriers and serial wrappers.
+            final WhereFilter copiedFilter = ExtractFilterWithoutBarriers.of(filter).copy();
```

Comment:
In the case of conjunctive / disjunctive filters, to the componentFilters need to be copied as well? I'm not sure exactly when WhereFilter needs to be copied / if it can be a "shallow" copy or needs to be deep.

Reply:
Conjunctive / Disjunctive filters do a deep copy:
```
    @Override
    public ConjunctiveFilter copy() {
        return new ConjunctiveFilter(WhereFilter.copyFrom(getComponentFilters()));
    }
```

Reply:
Should we handle the copy-taking in `ExtractFilterWithoutBarriers` so the top-level caller doesn't need to be responsible? Should be more efficient impl (/ less extraneous `io.deephaven.engine.liveness.LivenessManager#manage` as well).